compiler/include/stmt.h

```diff
@@ -275,6 +276,46 @@ class GotoStmt : public Stmt {
   LabelSymbol*        gotoTarget()                                     const;
 };
 
+
+/************************************ | *************************************
+*                                                                           *
+*                                                                           *
+************************************* | ************************************/
+
+//
+// forall loop statement
+//
+class ForallStmt : public Stmt
+{
+public:
+  // All pointer fields are always non-null.
+  BlockStmt*     fsBlock;   // contains defs for index variable(s) and fBody
+  AList          fIter;     // iterable expression(s)
+  bool           fZippered;
+  ForallIntents* fWith;     // todo: replace with AList of per-variable "clauses"
+
+  // derived from the above
+  VarSymbol* fIdxVar()     const;
+  VarSymbol* fIdxCopyVar() const;
+  BlockStmt* fBody()       const; // forall loop body
```

Comment:
Also, I'd like this information to still work after the Forall statements are "lowered" in resolution. But that is something we can discuss if it's difficult.